Documentation/kubernetes-networking.md

```diff
@@ -0,0 +1,101 @@
+# Kubernetes Networking
+
+## Network Model
+
+The Kubernetes network model outlines four methods of component communication:
+
+* Container-to-Container Communication
+    * Containers launched within the same pod can all reach each other's ports via localhost. This simplifies / secures communication between applications launched in the same Pod.
+
+* Pod-to-Pod Communication
+    * Each Pod in a Kubernetes cluster is assigned an IP in a flat shared networking namespace. This allows for a clean network model where Pods, from a networking perspective, can be treated much like VMs or physical hosts.
+
+* Pod-to-Service Communication
+    * Services are implemented by assigning Virtual IPs which clients can access and are transparently proxied to the Pods grouped by that service. Requests to the Service IPs are intercepted by a kube-proxy process running on all nodes, which is then responsible for routing to the correct Pod.
+
+* External-to-Internal Communication
+    * Accessing services from outside the cluster is generally implemented by configuring external loadbalancers which target all nodes in the cluster. Once traffic arrives at a node, it is routed to the correct Service backends via the kube-proxy.
```

Comment:
hmm, I thought that only nodes hosting qualified pods would be added as backends to an ELB, but I can't seem to find that in code